pkg/pillar/cmd/zedmanager/updatestatus.go

```diff
@@ -1032,6 +1042,43 @@ func doActivate(ctx *zedmanagerContext, uuidStr string,
 	return changed
 }
 
+// Check if VifUsed has changed and return true if it has
+func updateVifUsed(statusPtr *types.AppInstanceStatus, ds types.DomainStatus) bool {
+	changed := false
+	for i := range statusPtr.UnderlayNetworks {
+		ulStatus := &statusPtr.UnderlayNetworks[i]
+		net := lookupVifInfo(ds, ulStatus.Vif)
+		if net != nil && net.VifUsed != ulStatus.VifUsed {
+			log.Infof("Found VifUsed %s for Vif %s", net.VifUsed, ulStatus.Vif)
+			ulStatus.VifUsed = net.VifUsed
+			changed = true
+		}
+	}
+	for i := range statusPtr.OverlayNetworks {
+		olStatus := &statusPtr.OverlayNetworks[i]
+		net := lookupVifInfo(ds, olStatus.Vif)
+		if net != nil && net.VifUsed != olStatus.VifUsed {
+			log.Infof("Found VifUsed %s for Vif %s", net.VifUsed, olStatus.Vif)
+			olStatus.VifUsed = net.VifUsed
+			changed = true
+		}
+	}
+
+	// for _, ec := range config.OverlayNetworkList {
+	return changed
+}
+
+// Lookup based on the name aka Vif
+func lookupVifInfo(ds types.DomainStatus, vif string) *types.VifInfo {
```

Comment:
This sounds like a good function to be a method of domainSTatus

ds.VifInfoByName(vif)

Reply:
Since the "name" is in the Vif field I'll name it ds.VifInfoByVif()